cmake/tpl/SundialsLapack.cmake

```
@@ -26,41 +26,34 @@
 # Section 1: Include guard
 # -----------------------------------------------------------------------------
 
-if(NOT DEFINED SUNDIALS_LAPACK_INCLUDED)
-  set(SUNDIALS_LAPACK_INCLUDED)
-else()
-  return()
-endif()
+include_guard(GLOBAL)
 
 # -----------------------------------------------------------------------------
 # Section 2: Check to make sure options are compatible
 # -----------------------------------------------------------------------------
 
 # LAPACK does not support extended precision
 if(ENABLE_LAPACK AND SUNDIALS_PRECISION MATCHES "EXTENDED")
-  print_error("LAPACK is not compatible with ${SUNDIALS_PRECISION} precision")
+  message(
+    FATAL_ERROR "LAPACK is not compatible with ${SUNDIALS_PRECISION} precision")
 endif()
 
 # -----------------------------------------------------------------------------
 # Section 3: Find the TPL
 # -----------------------------------------------------------------------------
 
-# If LAPACK libraries are undefined, try to find them.
-if(NOT LAPACK_LIBRARIES)
-  find_package(LAPACK REQUIRED)
-endif()
+find_package(LAPACK REQUIRED)
 
-# If we have the LAPACK libraries, display progress message.
-if(LAPACK_LIBRARIES)
-  message(STATUS "Looking for LAPACK libraries... OK")
-  set(LAPACK_FOUND TRUE)
+# get path to LAPACK library to use in generated makefiles for examples, if
+# LAPACK_LIBRARIES contains multiple items only use the path of the first entry
+list(LENGTH LAPACK_LIBRARIES len)
+if(len EQUAL 1)
+  get_filename_component(LAPACK_LIBRARY_DIR ${LAPACK_LIBRARIES} PATH)
 else()
-  message(STATUS "Looking for LAPACK libraries... FAILED")
-  set(LAPACK_FOUND FALSE)
+  list(GET LAPACK_LIBRARIES 0 TMP_LAPACK_LIBRARIES)
+  get_filename_component(LAPACK_LIBRARY_DIR ${TMP_LAPACK_LIBRARIES} PATH)
 endif()
```

Untested suggestion:
```suggestion
list(GET LAPACK_LIBRARIES 0 TMP_LAPACK_LIBRARIES)
get_filename_component(LAPACK_LIBRARY_DIR ${TMP_LAPACK_LIBRARIES} PATH)
```